WebCore::FindOptionFlag should be an enum class

aprotyas opened the pull request:
#### 1a5937cad8704a64d6edcc79ccab066060d53809
<pre>
WebCore::FindOptionFlag should be an enum class
<a href="https://bugs.webkit.org/show_bug.cgi?id=271941">https://bugs.webkit.org/show_bug.cgi?id=271941</a>
<a href="https://rdar.apple.com/125667224">rdar://125667224</a>

Reviewed by Ryosuke Niwa and Charlie Wolfe.

This is a mechanical patch that &quot;enum class&quot;-ifies
WebCore::FindOptionFlag, and takes the liberty to rename the enum to
WebCore::FindOption, which I think is a more fitting name considering
how enum class cases are referenced.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::rangeOfStringClosestToRangeInDirection const):
(WebCore::Accessibility::isAccessibilityTextSearchMatch):
* Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp:
(WebCore::FragmentDirectiveRangeFinder::findRangeFromNodeList):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::findString):
(WebCore::start):
(WebCore::end):
(WebCore::makeBoundaryPointAfterNodeContents):
(WebCore::makeBoundaryPointAfterNode):
(WebCore::Editor::rangeOfString):
(WebCore::Editor::countMatchesForText):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::executeFindString):
* Source/WebCore/editing/FindOptions.h:
(): Deleted.
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::SearchBuffer::SearchBuffer):
(WebCore::SearchBuffer::isWordStartMatch const):
(WebCore::SearchBuffer::search):
(WebCore::rangeForMatch):
(WebCore::findClosestPlainText):
(WebCore::findPlainText):
* Source/WebCore/editing/TextIterator.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::find const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::findString):
(WebCore::Page::findTextMatches):
(WebCore::Page::rangeOfString):
* Source/WebCore/testing/Internals.cpp:
(WebCore::parseFindOptions):
* Source/WebKit/Shared/WebFindOptions.cpp:
(WebKit::core):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::countFindMatches):
(WebKit::PDFPlugin::findString):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::countFindMatches):
(WebKit::compareOptionsForFindOptions):
(WebKit::UnifiedPDFPlugin::findString):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::addDictationAlternative):
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::findString):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::adjustCandidateAutocorrectionInFrame):
(WebKit::WebPage::applyAutocorrectionInternal):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(coreOptions):

Canonical link: <a href="https://commits.webkit.org/276863@main">https://commits.webkit.org/276863@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2213e9496903bba77f048bf720a48cee0884896

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/45998 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25127 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48583 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48669 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42038 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48305 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29437 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22525 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37617 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46576 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22210 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39676 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18808 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19607 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40772 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4042 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42303 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41123 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50451 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/20994 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17471 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44771 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22294 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43663 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10190 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22653 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/21988 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
```diff
@@ -1333,15 +1333,15 @@ static bool getEventTypeFromWebEvent(const WebEvent& event, NSEventType& eventTy
-    bool wrapSearch = options.contains(WebCore::WrapAround);
+    bool searchForward = !options.contains(FindOption::Backwards);
+    bool caseSensitive = !options.contains(FindOption::CaseInsensitive);
+    bool wrapSearch = options.contains(WebCore::FindOption::WrapAround);
```
```suggestion
    bool wrapSearch = options.contains(FindOption::WrapAround);
```

aprotyas replied:
Fixed in f2213e9

aprotyas committed:
WebCore::FindOptionFlag should be an enum class
https://bugs.webkit.org/show_bug.cgi?id=271941
rdar://125667224

Reviewed by Ryosuke Niwa and Charlie Wolfe.

This is a mechanical patch that "enum class"-ifies
WebCore::FindOptionFlag, and takes the liberty to rename the enum to
WebCore::FindOption, which I think is a more fitting name considering
how enum class cases are referenced.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::rangeOfStringClosestToRangeInDirection const):
(WebCore::Accessibility::isAccessibilityTextSearchMatch):
* Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp:
(WebCore::FragmentDirectiveRangeFinder::findRangeFromNodeList):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::findString):
(WebCore::start):
(WebCore::end):
(WebCore::makeBoundaryPointAfterNodeContents):
(WebCore::makeBoundaryPointAfterNode):
(WebCore::Editor::rangeOfString):
(WebCore::Editor::countMatchesForText):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::executeFindString):
* Source/WebCore/editing/FindOptions.h:
(): Deleted.
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::SearchBuffer::SearchBuffer):
(WebCore::SearchBuffer::isWordStartMatch const):
(WebCore::SearchBuffer::search):
(WebCore::rangeForMatch):
(WebCore::findClosestPlainText):
(WebCore::findPlainText):
* Source/WebCore/editing/TextIterator.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::find const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::findString):
(WebCore::Page::findTextMatches):
(WebCore::Page::rangeOfString):
* Source/WebCore/testing/Internals.cpp:
(WebCore::parseFindOptions):
* Source/WebKit/Shared/WebFindOptions.cpp:
(WebKit::core):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::countFindMatches):
(WebKit::PDFPlugin::findString):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::countFindMatches):
(WebKit::compareOptionsForFindOptions):
(WebKit::UnifiedPDFPlugin::findString):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::addDictationAlternative):
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::findString):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::adjustCandidateAutocorrectionInFrame):
(WebKit::WebPage::applyAutocorrectionInternal):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(coreOptions):

Canonical link: https://commits.webkit.org/276863@main
